src/main/java/decodes/cwms/algo/ResEvapAlgo.java

```diff
@@ -0,0 +1,526 @@
+		//todo change resevap rating
+		RatingValue[] rt = rate0.getRatingValues();
+
+		// Use area units to determine unit system
+		String[] runits = rate0.getRatingUnits();
+		boolean is_english = false;
+//		unit_system = Units.getUnitSystemForUnits(runits[1]);
+//		if (unit_system == Units.ENGLISH_ID){
+//			is_english = true;
+//		}
+
+		double[] elev_array = new double[rcount];
+		double[] area_array = new double[rcount];
+		for ( int i =0; i< rcount; i++){
+			RatingValue rv = rt[i];
+			elev_array[i] = rv.getIndValue();
+			area_array[i] = rv.getDepValue();
+		}
+		reservoir.setElevAreaCurve(elev_array, area_array, rcount, is_english);
+
+
+
+		reservoir.setInstrumentHeights(32.81, 32.81, 32.81);
+
+		//double startElev = getStartElevation();
+
+		reservoir.setElevationTs(ElevTS);
+		reservoir.setElevation(ElevTS.findNextIdx(_timeSliceBaseTime));
+		reservoir.setZeroElevation(Zero_elevation);
+
+		resEvap = new ResEvap();
+
+		if (!resEvap.setReservoir(reservoir)) {
+			throw new DbCompException("Reservoir not in Database. Exiting Script.");
+		}
+
+		// reservoir structure setup so now set wtemp profile
+		int resj = reservoir.getResj();
+
+//		print("ResJ: " + str(resj));
+//		print("WTP len: " + str(len(wtp)));
+		double[] wtp;
+		try{
+			wtp = getProfiles();
+		}
+		catch(Exception ex){
+			throw new DbCompException("failed to load profiles");
+		}
+		// reverse array order
+		double[] wtpR = new double[resj];
+		for (int i = 0; i<resj+1; i++){
+			wtpR[i] = wtp[resj-i];
+		}
+
+		reservoir.setInitWaterTemperatureProfile(wtpR, resj);
+
+		resEvap._metData = metData;
+//AW:BEFORE_TIMESLICES_END
+	}
+
+	/**
+	 * Do the algorithm for a single time slice.
+	 * AW will fill in user-supplied code here.
+	 * Base class will set inputs prior to calling this method.
+	 * User code should call one of the setOutput methods for a time-slice
+	 * output variable.
+	 *
+	 * @throws DbCompException (or subclass thereof) if execution of this
+	 *        algorithm is to be aborted.
+	 */
+	protected void doAWTimeSlice()
+            throws DbCompException {
+//AW:TIMESLICE
+		try {
+			boolean noProblem = resEvap.compute(_timeSliceBaseTime, GMT_Offset);
+			if (!noProblem){
+				throw new DbCompException("ResEvap Compute Not Successful. Exiting Script.");
+			}
+		}
+		catch(Exception ex){
+			throw new DbCompException("ResEvap Compute Not Successful. Exiting Script.", ex);
+		}
+
+		List<Double> computedList = resEvap.getComputedMetTimeSeries();
+		setOutput(HourlySurfaceTemp, computedList.get(0));
+		setOutput(HourlySensible, computedList.get(1));
+		setOutput(HourlyLatent, computedList.get(2));
+		setOutput(HourlySolar, computedList.get(3));
+		setOutput(HourlyFluxIn, computedList.get(4));
+		setOutput(HourlyFluxOut, computedList.get(5));
+		setOutput(HourlyEvap, computedList.get(6));
+
+		setProfiles(hourlyWTP, resEvap.getHourlyWaterTempProfile(), _timeSliceBaseTime);
+
+		//TODO Aggregation
+		if(_timeSliceBaseTime.getDate() != LastDate.getDate()){
```

Comment:
Move to afterTimeSlice and you can ditch the date logic, with the aggregating algorithm you can be sure that the output will be put at the correct timestamp according to user specifications.